services/provider/server/server.go

```diff
@@ -550,16 +554,25 @@ func (s *OCSProviderServer) GetStorageClaimConfig(ctx context.Context, req *pb.S
 	}
 	var extR []*pb.ExternalResource
 
-	var storageClassName string
-	storageClassData := map[string]string{}
+	storageRequestHash := getStorageRequestHash(req.StorageConsumerUUID, req.StorageClaimName)
+	provisionerSecretName := storageClaimCephCsiSecretName("provisioner", storageRequestHash)
+	nodeSecretName := storageClaimCephCsiSecretName("node", storageRequestHash)
 
+	storageClassData := map[string]string{
+		"csi.storage.k8s.io/provisioner-secret-name":       provisionerSecretName,
+		"csi.storage.k8s.io/node-stage-secret-name":        nodeSecretName,
+		"csi.storage.k8s.io/controller-expand-secret-name": provisionerSecretName,
+	}
```

Comment:
Why are you assuming these secrets are in the reply and precalculate them? You should only invoke these based on reply content. And you are already doing so inside the switch statement

Reply:
Actually, this was also my first comment to Jose's PR and later realised this block will only be hit after the StorageRequest phase is READY by which time it is guaranteed the secrets will exist.

Reply:
changed code to explicitly check for the secret existence. done.